sktime/forecasting/croston.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import pandas as pd
+from sktime.forecasting.base._sktime import _SktimeForecaster
+from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+
+
+DEFAULT_ALPHA = 0.05
+
+
+class Croston(_OptionalForecastingHorizonMixin, _SktimeForecaster):
+    """
+    Implementation of Croston's Method
+    ----------------------------------
+    Useful for Forecasting Intermittent Demand Time Series.
+    The Croston() function produces forecasts using Croston’s method.
+    It simply uses α = 0.1 by default,
+    and p = 0 is set to be equal to the first observation in each of the series.
+    This is consistent with the way Croston envisaged the method being used.
+
+    Parameters:
+    -----------
+        demand: array-like
+            Historical data
+        future_periods: int
+            Time period for which predictions are required
+        alpha: float, optional(default=0.1)
+            Smoothing parameter
+
+    Returns:
+    --------
+        forecast: array-like
+            Forecasted demand (on average per period) diff."""
+
+    def __init__(self, alpha=DEFAULT_ALPHA):
+        # hyperparameter
+        self.alpha = alpha
```

Comment:
Could we rename `alpha` to avoid ambiguity with the alpha parameter for confidence intervals? `smoothing` should work.

Reply:
Did that, but got `TypeError: predict() got an unexpected keyword argument 'alpha'` from 20 failed unit tests

Reply:
Not sure where the error comes from. Perhaps push the latest changes to see what's going on?

Reply:
Yep, I just pushed the changes. They'll throw these tests `TypeError: predict() got an unexpected keyword argument 'alpha'`